### PR TITLE
Implement script to parse provisioners duration from packer diagnostic logs

### PR DESCRIPTION
**Pull request by maxim-lobanov**
# Description
This PR adds script `measure-provisioners-duration.ps1` to repo. This script parses packer diagnostic logs and determine the duration of every step. As a result, it prints the following output:
- The whole build timeline 
image
- The Top 25 longest provisioners
image

All changes:
- Move logic to find template to the separate step on Windows / Linux to reuse it later and avoid code duplication
- Set `PACKER_LOG: 1`. image generation logs will still contain only short log. But diagnostic logs will be saved to separate file.

Build examples:
- https://dev.azure.com/github/virtual-environments/_build/results?buildId=90048&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=8b69240d-5eca-52e7-db53-21813d4d30a5
- https://dev.azure.com/github/virtual-environments/_build/results?buildId=90047&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=027aaa20-7755-5366-f0d4-a9dbdf9aa415

## Purpose of these changes:
- Find bottlenecks in image generation (what script take the most of the time)
- Analyze new PRs, when someone adds new software. We can take a look at logs of this step and check quickly how much time new script takes.

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images.CI/measure-provisioners-duration.ps1 [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+$provisionersList | ForEach-Object {
+    Write-Host "- $($_.Duration) | $($_.ProvisionerType)"
+    $_.SubItems | ForEach-Object {
+        Write-Host "      $($_.Duration) | $($_.Command)"
+    }
+    Write-Host ""
+}
+Write-Host "Total provisioners time: $totalProvisionersTime"
+
+if ($PrintTopNLongest -and ($PrintTopNLongest -gt 0)) {
```
Comment (al-cheb) [inline images.CI/measure-provisioners-duration.ps1:114]: nitpick: remove `$PrintTopNLongest -and` and ()

File: images.CI/measure-provisioners-duration.ps1 [hunk 2]
```diff
@@ -0,0 +1,121 @@
+param(
+    [string]$PackerLogPath = "$env:HOME/Downloads/temp2/packer-mac.txt",
```
Comment (AlenaSviridenko) [inline images.CI/measure-provisioners-duration.ps1:2]: why is it `packer-mac.txt`? should it be `packer-log.txt` instead?

File: images.CI/measure-provisioners-duration.ps1 [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+)
+
+$DateTimeRegex = "(\d+\/\d+\/\d+ \d+:\d+:\d+)"
+$TelemetryLineRegex = "\[INFO\] \(telemetry\)"
+$StartProvisionerRegex = "^${DateTimeRegex} ${TelemetryLineRegex} Starting provisioner (.+)$"
+$EndProvisionerRegex = "^${DateTimeRegex} ${TelemetryLineRegex} ending (.+)$"
+$ShellScriptSubItemRegex = "${DateTimeRegex} ui: ==> .+: Provisioning with \w+ script: (.+)"
+$DownloadUploadSubItemRegex = "${DateTimeRegex} ui: ==> .+: (Downloading .+|Uploading .+)"
+
+function New-ProvisionerItem {
```
Comment (AlenaSviridenko) [inline images.CI/measure-provisioners-duration.ps1:15]: Maybe rename to `Start-ProvisionerItem` and then `End-ProvisionerItem`?

File: images.CI/measure-provisioners-duration.ps1 (hunk 2 above)
Comment (maxim-lobanov) [reply images.CI/measure-provisioners-duration.ps1]: It should be empty by default and Mandatory.
I have done this change locally but forgot to do commit before push :(
Fixed

File: images.CI/measure-provisioners-duration.ps1 (hunk 1 above)
Comment (maxim-lobanov) [reply images.CI/measure-provisioners-duration.ps1]: Thank you, fixed!

**Review by AlenaSviridenko — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by MichaelDavidOrtega — COMMENTED** [review]
> images.CI/measure-provisioners-duration.ps1

**Pull request merged**